### PR TITLE
Charge Listing Fee From Sale Price

### PR DESCRIPTION
**Pull request by h0ngcha0**
Charge Listing Fee and Commission Fee from Sale Price.

This contains https://github.com/alephium/alephium-nft/pull/31,

### CODE REVIEW

**Review by Lbqds — APPROVED** [review]
LGTM 👍  , only two questions.

File: contracts/marketplace/nft_marketplace.ral [hunk 1]
```diff
@@ -57,38 +57,42 @@ Contract NFTMarketPlace(
...
         let nftListing = NFTListing(nftListingContractId)
         let buyer = callerAddress!()
-        let price = nftListing.getPrice()
         let tokenOwner = nftListing.getTokenOwner()
+        let price = nftListing.getPrice()
+        let commissionRateOfListing = nftListing.getCommissionRate()
+        let listingFeeOfListing = nftListing.getListingFee()
 
-        nftListing.buy{buyer -> ALPH: price}(buyer)
+        let priceAfterFee = checkListingPriceAfterFee(price, commissionRateOfListing, listingFeeOfListing)
```
Comment (Lbqds) [inline contracts/marketplace/nft_marketplace.ral:65]: Question:

1. If someone lists an NFT, then the marketplace updates the commission rate, causing the `checkListingPriceAfterFee` fails, does this means that the NFT seller has to update the price?
2. Do we also need to call `checkListingPriceAfterFee` in the `updateNFTPrice` function?

Comment (Lbqds) [reply contracts/marketplace/nft_marketplace.ral:65]: Sorry, please ignore the first question, I found that we saved the previous commission rate and listing price in the `NFTListing` contract.

Comment (h0ngcha0) [reply contracts/marketplace/nft_marketplace.ral:65]: Sounds great :+1:

**Pull request merged**